lib/evmone/baseline.cpp

```diff
@@ -69,13 +70,71 @@ inline evmc_status_code check_requirements(
     return EVMC_SUCCESS;
 }
 
-/// Dispatch the instruction currently pointed by "pc".
-#define DISPATCH() break  // Break out of switch statement.
 
-/// Increment "pc" and dispatch the instruction.
-#define DISPATCH_NEXT() \
-    ++code_it;          \
-    DISPATCH()
+/// Implementation of a generic instruction "case".
+#define DISPATCH_CASE(OPCODE)                                          \
+    case OPCODE:                                                       \
+        if (code_it = invoke(op2fn::OPCODE, state, code_it); !code_it) \
+            goto exit;                                                 \
+        break
+
+/// The signature of basic instructions which always succeed, e.g. ADD.
+using InstrFn1 = void(ExecutionState&) noexcept;
```

Comment:
Maybe more descriptive names for these types? `SucceedingFn`, `FailingFn`, `TerminatingFn`, `CodePositionFn` ?